### PR TITLE
Add min:0 to templates of Bookshelf DBs. Issue #8436

### LINKED ISSUE #8436: Database connection timeouts after some time and strapi crashed
### Describe the bug
I am using Strapi as dockerised app inside Caprover PAAS. What happens is that when some time of inactivity, the app crashed in the next request. It looks like that connection to database is terminated.

This happens with both mysql and posgresql backend.

The strapi crashes then. 

### Steps to reproduce the behavior
1. install strapi as dockerized app into Caprover PAAS
2. use one-click mysql or pgsql app

### Logs:
```
2020-10-22T11:20:22.986874066Z [2020-10-22T11:20:22.986Z] debug GET /blog-articles?_start=0&_limit=5&_sort=Date:DESC&IsPublished=true (23 ms) 200
2020-10-22T11:20:23.181097954Z [2020-10-22T11:20:23.180Z] debug GET /blog-articles/count?IsPublished=true (15 ms) 200
2020-10-22T11:46:05.945389043Z [2020-10-22T11:46:05.945Z] error Error: read ECONNRESET
2020-10-22T11:46:05.945432223Z     at TCP.onStreamRead (internal/stream_base_commons.js:208:20)
2020-10-22T11:46:05.945438229Z     --------------------
2020-10-22T11:46:05.945442289Z     at Protocol._enqueue (/src/app/node_modules/mysql/lib/protocol/Protocol.js:145:48)
2020-10-22T11:46:05.945446377Z     at Connection.query (/src/app/node_modules/mysql/lib/Connection.js:208:25)
2020-10-22T11:46:05.945450155Z     at /src/app/node_modules/knex/lib/dialects/mysql/index.js:135:18
2020-10-22T11:46:05.945453786Z     at new Promise (<anonymous>)
2020-10-22T11:46:05.945458024Z     at Client_MySQL._query (/src/app/node_modules/knex/lib/dialects/mysql/index.js:129:12)
2020-10-22T11:46:05.945461647Z     at Client_MySQL.query (/src/app/node_modules/knex/lib/client.js:169:17)
2020-10-22T11:46:05.945465213Z     at Runner.query (/src/app/node_modules/knex/lib/runner.js:151:36)
2020-10-22T11:46:05.945468775Z     at /src/app/node_modules/knex/lib/runner.js:40:23
2020-10-22T11:46:05.945472593Z     at /src/app/node_modules/knex/lib/runner.js:277:24
2020-10-22T11:46:05.946054687Z [2020-10-22T11:46:05.945Z] debug GET /blog-articles?_start=0&_limit=5&_sort=Date:DESC&IsPublished=true (7 ms) 500
2020-10-22T11:48:21.500096306Z [2020-10-22T11:48:21.499Z] error Error: read ECONNRESET
2020-10-22T11:48:21.500134178Z     at TCP.onStreamRead (internal/stream_base_commons.js:208:20)
2020-10-22T11:48:21.500139415Z     --------------------
2020-10-22T11:48:21.500143792Z     at Protocol._enqueue (/src/app/node_modules/mysql/lib/protocol/Protocol.js:145:48)
2020-10-22T11:48:21.500149650Z     at Connection.query (/src/app/node_modules/mysql/lib/Connection.js:208:25)
2020-10-22T11:48:21.500153608Z     at /src/app/node_modules/knex/lib/dialects/mysql/index.js:135:18
2020-10-22T11:48:21.500164357Z     at new Promise (<anonymous>)
2020-10-22T11:48:21.500168736Z     at Client_MySQL._query (/src/app/node_modules/knex/lib/dialects/mysql/index.js:129:12)
2020-10-22T11:48:21.500172376Z     at Client_MySQL.query (/src/app/node_modules/knex/lib/client.js:169:17)
2020-10-22T11:48:21.500176009Z     at Runner.query (/src/app/node_modules/knex/lib/runner.js:151:36)
2020-10-22T11:48:21.500179811Z     at /src/app/node_modules/knex/lib/runner.js:40:23
2020-10-22T11:48:21.500183483Z     at /src/app/node_modules/knex/lib/runner.js:277:24
```

### System

- Node.js version: 12 LTS
- NPM version:
- Strapi version: 3.2.4
- Database: mysql or pgsql
- Operating system: docker - using node:lts-alpine

### PR DESCRIPTION
**Pull request by NgyAnthony**
### What does it do?

Set number of minimum connections always available by Knex from `min:2`(default) to` min: 0` for all bookshelf databases.

### Why is it needed?

Knex needs 2 connections always available by default, however Docker Swarm will close any inactive connection within 15min. Having those connections closed will throw an `error Error: read ECONNRESET` error on MySQL. To fix this we set the default connections always available needed by Knex to 0. 

### Related issue(s)/PR(s)

Fix #8436

### CODE REVIEW

**Comment by alexandrebodin** [comment]
Hi @NgyAnthony  thank you for this PR. Instead of adding it to the templates (which won't fix the current existing apps) I think we can directly set it as default  in the code too in this file packages/strapi-connector-bookshelf/lib/knex.js

what do you think ?

**Comment by NgyAnthony** [comment]
> Hi @NgyAnthony thank you for this PR. Instead of adding it to the templates (which won't fix the current existing apps) I think we can directly set it as default in the code too in this file packages/strapi-connector-bookshelf/lib/knex.js
> 
> what do you think ?

Sure, sounds good to me.
Modified knex.js, if you could point me in the right direction if I had to modify other stuff that would be great 👍

File: packages/strapi-connector-bookshelf/lib/knex.js [hunk 1]
```diff
@@ -101,7 +101,9 @@ module.exports = strapi => {
           },
           ...connection.options,
           debug: _.get(connection.options, 'debug', false),
-          min: _.get(connection.options.pool, 'min', 0),
+          pool: {
```
Comment (alexandrebodin) [inline packages/strapi-connector-bookshelf/lib/knex.js:104]: nice I didn't even notice the old param that wasn't in the right place 👍

**Review by alexandrebodin — APPROVED** [review]

**Comment by Zn4rK** [comment]
This makes it impossible to actually configure the pool apart from the min value. The property "pool" from `connection.options` is always overridden by the new property. `connection.options` needs to be spread below the defaults.

**Comment by NgyAnthony** [comment]
Welp. This small PR sure brings a lot of trouble, never knew you had to be this careful to configure stuff, very sorry about that.

You might want to revert #9348 @alexandrebodin in the meantime.
Previous commit indeed fixed an issue where min and max were explicitly specified https://github.com/strapi/strapi/commit/149c1f7737c4a06dcfe5675b3106f89801045283

**Pull request merged**